authors/apps/authentication/views.py

```diff
@@ -73,3 +75,48 @@ def update(self, request, *args, **kwargs):
 
         return Response(serializer.data, status=status.HTTP_200_OK)
 
+class SocialSignUp(CreateAPIView):
+    permission_classes = (AllowAny,)
+    renderer_classes = (UserJSONRenderer,)
+    serializer_class = UserSerializer
+
+    def create(self, request,  *args, **kwargs):
+        # Get access token and create user or authenticate
+
+        provider = request.data['provider']
+
+        authed_user = request.user if not request.user.is_anonymous else None
+
+        strategy = load_strategy(request)
+
+        backend = load_backend(strategy=strategy, name=provider, redirect_uri=None)
+
+        if isinstance(backend, BaseOAuth2):
+            # Grab the access_token
+            token = request.data['access_token']
```

Comment:
Handle error if this is not provided to.

Reply:
noted.